roles/satellite-clone/tasks/main.yml

```diff
@@ -254,15 +266,17 @@
   - name: Reset admin password
     command: foreman-rake permissions:reset password=changeme
 
-  - name: Setup hammer for foreman-maintain (satellite 6.4+ only)
+  - name: Setup hammer for foreman-maintain
     lineinfile:
       path: /etc/foreman-maintain/foreman-maintain-hammer.yml
       state: present
       create: yes
       line: "---\n:foreman:\n  :username: admin\n  :password: changeme"
     when: satellite_version in ["6.4", "6.5"]
 
-  - name: Run installer upgrade (satellite 6.2+ only)
+  # why exactly do we run the installer upgrade here? would like to comment to make it obvious
+  # my thought is that if we start with an empty machine we should get the latest stuff in the first installer run?
```

Comment:
This is for the z-version differences that I mentioned in the other comment:
> Sometimes you get these differences with different z-versions. For example, the backup is from 6.3.1 and we are installing the latest 6.3, 6.3.5

This ensures all the databases are migrated and any other expected configuration is present that may not have been present in an earlier z-version backup.

If you feel up to adding comments around these things it would be great to let other devs know these things. Feel free to alter the step name as well, this is user-facing and could provide some better info as the user runs it.

Thanks for bringing this up!